src/Middleware/ThrottleMiddleware.php

```diff
@@ -40,6 +41,13 @@ public function __construct($config = [])
      */
     public function __invoke(ServerRequest $request, Response $response, callable $next)
     {
+        $event = $this->dispatchEvent(self::EVENT_BEFORE_THROTTLE, [
```

Comment:
`implements EventDispatcherInterface` should also be added to the middleware/filter classes.